packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1263,6 +1263,12 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
 
   private readonly domain: CfnDomain;
 
+
+  private accessPolicyStatements: iam.PolicyStatement[] = []
```

Comment:
Again, my apologies if I was unclear previously. This property should be on `ElasticsearchAccessPolicy`; there is no need for `Domain` to track this state